Makefile

```diff
@@ -145,39 +156,59 @@ run-% : $(OBJDIR)/%
 	@tests/tap.sh $(<:build/%=%)
 
 test : $(tests:$(OBJDIR)/%=run-%) $(examples:$(OBJDIR)/%=run-%)
+# run test target in parallel
+tst : ;@$(MAKE) $(MFLAGS) V=$(V) test
+# CPU C tests only for backend %
+ctc-% : $(ctests);@$(foreach tst,$(ctests),$(tst) /cpu/$*;)
 
 prove : $(tests) $(examples)
 	$(PROVE) $(PROVE_OPTS) --exec 'tests/tap.sh' $(tests:$(OBJDIR)/%=%) $(examples:$(OBJDIR)/%=%)
+# run prove target in parallel
+prv : ;@$(MAKE) $(MFLAGS) V=$(V) prove
 
 examples : $(examples)
 
 $(ceed.pc) : pkgconfig-prefix = $(abspath .)
 $(OBJDIR)/ceed.pc : pkgconfig-prefix = $(prefix)
-.INTERMEDIATE: $(OBJDIR)/ceed.pc
+.INTERMEDIATE : $(OBJDIR)/ceed.pc
 %/ceed.pc : ceed.pc.template | $$(@D)/.DIR
 	@sed "s:%prefix%:$(pkgconfig-prefix):" $< > $@
 
+# OCCA env, OKL files cache, clear & rule to install
+OCCA_CACHE_DIR     ?= ${HOME}/.occa
+OCCA_LIB_CACHE_DIR := $(OCCA_CACHE_DIR)/libraries
+OKL_KERNELS        := $(shell find backends/occa -type f -name '*.okl')
+OKL_CACHED_KERNELS := $(subst backends/occa,$(OCCA_LIB_CACHE_DIR)/ceed,$(OKL_KERNELS))
+okl-cache: $(OKL_CACHED_KERNELS)
+okl-clear: ;@occa clear -y -l ceed
+$(OCCA_LIB_CACHE_DIR)/ceed/%.okl: backends/occa/*.okl
+	echo find . -type f -name $*.okl
+	@echo "Caching: $(shell find . -type f -name $*.okl)"
+	@occa cache ceed $(shell find . -type f -name $*.okl)
+
 install : $(libceed) $(OBJDIR)/ceed.pc
-	$(INSTALL) -d "$(DESTDIR)$(includedir)" "$(DESTDIR)$(libdir)" "$(DESTDIR)$(pkgconfigdir)"
+	$(INSTALL) -d "$(DESTDIR)$(includedir)" "$(DESTDIR)$(libdir)" "$(DESTDIR)$(okldir)" "$(DESTDIR)$(pkgconfigdir)"
 	$(INSTALL_DATA) include/ceed.h "$(DESTDIR)$(includedir)/"
 	$(INSTALL_DATA) $(libceed) "$(DESTDIR)$(libdir)/"
 	$(INSTALL_DATA) $(OBJDIR)/ceed.pc "$(DESTDIR)$(pkgconfigdir)/"
+	$(INSTALL_DATA) $(OKL_KERNELS) "$(DESTDIR)$(okldir)/"
+
+.PHONY : all cln clean print test tst prove prv examples style install doc okl-cache okl-clear
 
-.PHONY: all cln clean print test tst examples astyle install doc
 cln clean :
 	$(RM) *.o *.d $(libceed)
 	$(RM) -r *.dSYM $(OBJDIR) $(LIBDIR)/pkgconfig
 	$(MAKE) -C examples clean
 	$(MAKE) -C examples/mfem clean
-	cd examples/nek5000; bash make-nek-examples.sh clean; cd ../..;
+	(cd examples/nek5000 && ./make-nek-examples.sh clean)
```

Comment:
So the shebang in that file is not the first line, thus is ignored, so you might not get bash. What was wrong with calling it `bash make-nek-examples.sh clean`?

Reply:
This line can be left as it was I agree; but the 'clean' target has to be push before the 'Set defaults for the parameter' and exit '0', instead of exiting with the last call return code